### PR TITLE
Local language support (LC_ALL) added to installation& launcher on macOS

### LINKED ISSUE #1880: INSTALLATION on chinese computer
### Prerequisites

* [ ] Did you check past issues for similar issue(s) ?
* [ ] Can you reproduce the issue with any of the files found in `sct_testing_data` ?
* [ ] If you are using a development environment, is your environment updated ?

### Description

Cannot import tensorflow on a chinese computer. 
Solution: run `export LC_ALL=en_US.UTF-8` prior to installing and using deeplearning methods
see https://github.com/explosion/spaCy/issues/1517

### Steps to Reproduce

1. ./sct_install

### State of spinalcordtoolbox

version 3.2.1

### PR DESCRIPTION
**Pull request by abtahizadeh**
### Requirements

* [* ] Is there a issue open describing the motivation for this pull request ?
* [ ] Does the pull request need an update of the documentation ?
* [ *] Does the pull request need additional test(s) ?
Tested on macOS
* [ ] Clean up the warnings generated by the [style guide](https://www.jetbrains.com/help/pycharm/2017.1/code-inspection.html)

### Description of the Change

Adding support for non-English language local Unicode during the installation.
Please note that `LC_ALL` overwrites `LANG`

### Steps and Constraints

`install_sct` updated

### Applicable Issues

Implements or Fixes #1880  and #1851 (duplicate)

### CODE REVIEW

**Comment by perone** [comment]
We talked about that in Slack before, I don't think it's a good solution because it will be forcing English locale into users computers. It will instruct internationalized programs to use the English language even if their language isn't English.

**Comment by abtahizadeh** [comment]
@perone Didn't know about your previous discussion, I don't think there's other solution than this ... what should we do?

@tanguyduval thanks! update in progess

**Comment by zougloub** [comment]
Setting the ``en_US.UTF-8`` locale globally in the bashrc is *not* the proper solution. A more localized fix is needed. The closest I could think of was in the ``sct_launcher``.

**Comment by abtahizadeh** [comment]
@zougloub normally, `$LC_ALL` is not set by default in macOS, which I guess wouldn't hurt to modify it. I agree with taking an action on `sct_launcher`, printing warning might not be enough IMO.

**Comment by abtahizadeh** [comment]
Moving the solution to `sct_launcher`

**Comment by abtahizadeh** [comment]
Works on my MAC, @tanguyduval can you please verify?

**Comment by tanguyduval** [comment]
Right, crash appears during installation. Lc_all should be define at the
begining of install_sct also.

I cannot check this Pr because I have no more access to a foreign computer.

**Comment by abtahizadeh** [comment]
@jcohenadad done
@tanguyduval Thanks for the tips
@zougloub or @perone Can you plz approve?

File: install_sct [hunk 1]
```diff
@@ -239,6 +239,16 @@ fi
 # Fetch OS type
 if uname -a | grep -i  darwin > /dev/null 2>&1; then
   # OSX
+
+  # Fix for non-English Unicode systems on MAC
+  if [ -z ${LC_ALL} ]; then
+    echo "export LC_ALL=en_US.UTF-8" >> ~/.bash_profile
```
Comment (jcohenadad) [inline install_sct]: i though we agreed to not modify the bash_profile?  how about updating the env variable during install (to make TF happy), and also modify it in the launcher (to have it run during execution of TF-related functions)?

Comment (abtahizadeh) [reply install_sct]: done
I am not able to test, struggling with some problems related to pip ...

File: install_sct [hunk 2]
```diff
@@ -242,11 +242,11 @@ if uname -a | grep -i  darwin > /dev/null 2>&1; then
 
   # Fix for non-English Unicode systems on MAC
   if [ -z ${LC_ALL} ]; then
-    echo "export LC_ALL=en_US.UTF-8" >> ~/.bash_profile
+    export LC_ALL=en_US.UTF-8
```
Comment (jcohenadad) [inline install_sct]: not going to work with (t)csh.

**Comment by jcohenadad** [comment]
@abtahizadeh have you been able to test the modifications you propose on a laptop where the error occurs?

File: install_sct (hunk 2 above)
Comment (jcohenadad) [reply install_sct]: Can it enter this if statement in Linux system? (If not, then disregard my comment about csh)

Comment (abtahizadeh) [reply install_sct]: We have an `IF` statement to check, we execute only on macOS:
`if uname -a | grep -i  darwin > /dev/null 2>&1; then`

Comment (jcohenadad) [reply install_sct]: sorry the previous lines were hidden by github. All good then 👍

Comment (abtahizadeh) [reply install_sct]: @jcohenadad plz approve

Comment (jcohenadad) [reply install_sct]: @abtahizadeh as mentioned in https://github.com/neuropoly/spinalcordtoolbox/pull/1881#discussion_r197626836, this PR needs to be tested before merging. You mentioned you had issues with pip that you will be fixing.

File: install_sct (hunk 1 above)
Comment (jcohenadad) [reply install_sct]: @abtahizadeh changing local language to reproduce the error is extremely easy and takes 5s. Explanation here: https://github.com/Microsoft/vscode/issues/7301

**Comment by jcohenadad** [comment]
tested ea6ef085316110fc0361c82525985b19af1dbcb6 on an OSX Terminal configured with Japanese encoding. Installed passed successfully, however `sct_testing` behaves strangely and got stuck at `sct_get_centerline`:

~~~
sct_testing -d 1

--
Spinal Cord Toolbox (feature/1880/ea6ef085316110fc0361c82525985b19af1dbcb6)
Running /Users/julien/code/sct/scripts/sct_testing.py -d 1

Downloading testing data...

Trying URL: https://osf.io/z8gaj/?action=download
Downloading 20180125_sct_testing_data.zip...
Status: 100%|##############################| 6.78M/6.78M [00:04<00:00, 1.63MB/s]

Check if folder already exists...
WARNING: Folder sct_testing_data already exists. Removing it...
rm -rf sct_testing_data

Unzip data to: /Users/julien

Remove temporary file...
Done!


Path to testing data: /Users/julien/sct_testing_data
Checking sct_analyze_lesion.........................[OK]
Checking sct_analyze_texture........................[OK]
Checking sct_apply_transfo..........................[OK]
Checking sct_compute_ernst_angle....................[OK]
Checking sct_compute_hausdorff_distance.............The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
[OK]
Checking sct_compute_mtr............................[OK]
Checking sct_compute_mscc...........................[OK]
Checking sct_compute_snr............................[OK]
Checking sct_concat_transfo.........................[OK]
Checking sct_convert................................[OK]
Checking sct_create_mask............................The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
The process has forked and you cannot use this CoreFoundation functionality safely. You MUST exec().
Break on __THE_PROCESS_HAS_FORKED_AND_YOU_CANNOT_USE_THIS_COREFOUNDATION_FUNCTIONALITY___YOU_MUST_EXEC__() to debug.
[OK]
Checking sct_crop_image.............................[OK]
Checking sct_dice_coefficient.......................[OK]
Checking sct_deepseg_gm.............................[OK]
Checking sct_deepseg_sc.............................[OK]
Checking sct_detect_pmj.............................[OK]
Checking sct_dmri_compute_dti.......................[OK]
Checking sct_dmri_concat_bvals......................[OK]
Checking sct_dmri_concat_bvecs......................[OK]
Checking sct_dmri_create_noisemask..................[OK]
Checking sct_dmri_compute_bvalue....................[OK]
Checking sct_dmri_moco..............................[OK]
Checking sct_dmri_separate_b0_and_dwi...............[OK]
Checking sct_dmri_transpose_bvecs...................[OK]
Checking sct_extract_metric.........................[OK]
Checking sct_fmri_compute_tsnr......................[OK]
Checking sct_fmri_moco..............................[OK]
Checking sct_get_centerline.........................
~~~

**Comment by abtahizadeh** [comment]
Tested on manufactured Sucheng (Chinese) macOS, installation successful,
got stuck at `sct_deepseg_gm` (!) when running `sct_testing -d 1` ...

screen shot 2018-07-03 at 2 32 11 pm

**Comment by perone** [comment]
@abtahizadeh is this Sucheng (Chinese) macOS running inside a VM or something like that ? The `sct_deepseg_gm` can be very slow if the VM is running without virtualization extensions. Did you check if the CPU is idle when the `sct_deepseg_gm` is stuck ? It might happen that it is just being slow.

**Comment by abtahizadeh** [comment]
@perone Yes, VM, and I agree that might be because of not having enough resources, I also suspected that it might be because of running out of RAM. I am currently running another test to see if it works with 16G rams and 4 CPUs.

**Comment by abtahizadeh** [comment]
Test passed, the issue was related to insufficient memory/CPU, however, this bug still introduced: #1900

**Pull request merged**